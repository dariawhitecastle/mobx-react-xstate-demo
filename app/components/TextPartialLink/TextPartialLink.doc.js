import { describe, PropTypes } from 'react-desc'

export const doc = TextPartialLink => {
  const DocumentedTextPartialLink = describe(TextPartialLink)
    .description('A TextPartialLink')
    .usage(
      `import { TextPartialLink } from 'components/TextPartialLink';
<TextPartialLink />`,
    )

  DocumentedTextPartialLink.propTypes = {
    intl: PropTypes.object.description('Injected intl.').isRequired,
    text: PropTypes.string.description('String to be displayed.').defaultValue(''),
    textProps: PropTypes.object
      .description('Props to be passed to the FormattedMessage component.')
      .defaultValue({}),
    textStyle: PropTypes.object
      .description('Style prop to be applied to the FormattedMessage component.')
      .defaultValue({}),
    linkedText: PropTypes.string.description('String to be displayed as a link.').defaultValue(''),
    linkedTextProps: PropTypes.object
      .description('Props to be passed to the Button component.')
      .defaultValue({}),
    linkedTextStyle: PropTypes.object
      .description('Style prop to be applied to the Button component.')
      .defaultValue({}),
    onClick: PropTypes.func
      .description('onClick method for the Button component.')
      .defaultValue({}),
  }

  return DocumentedTextPartialLink
}
