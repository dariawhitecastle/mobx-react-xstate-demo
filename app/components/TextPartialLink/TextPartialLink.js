/* eslint-disable react/prop-types */
import React from 'react'
import { injectIntl, FormattedMessage } from 'react-intl'

import { Button } from '../Button'
import { Text } from '../Text'

/**
 *
 * TextPartialLink
 *
 * Displays text with a part of it being a link.
 *
 */
const TextPartialLink = ({
  intl,
  text,
  textProps,
  textStyle,
  linkedText,
  linkedTextProps,
  linkedTextStyle,
  onClick,
}) => (
  <div>
    <FormattedMessage {...textProps} style={textStyle} values={{ name: `${text}` }}>
      {message => (
        <div>
          <Text>{message}</Text>
          <Button
            {...linkedTextProps}
            label={intl.formatMessage(linkedText)}
            onClick={onClick}
            style={linkedTextStyle}
          />
        </div>
      )}
    </FormattedMessage>
  </div>
)

TextPartialLink.propTypes = {}

export default injectIntl(TextPartialLink)
