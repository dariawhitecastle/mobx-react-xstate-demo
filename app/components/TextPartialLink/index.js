export { default as TextPartialLink } from './TextPartialLink'
