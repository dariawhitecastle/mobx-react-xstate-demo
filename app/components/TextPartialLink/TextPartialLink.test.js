/**
*
* Tests for TextPartialLink
*
* @see
https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
*
*/

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import TextPartialLink from './TextPartialLink'
import { DEFAULT_LOCALE } from '../../i18n'

const renderComponent = (linkedTextStyle, textStyle) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <TextPartialLink
        linkedText={{
          id: `plainButtonLabel`,
          defaultMessage: 'Test Link',
        }}
        linkedTextStyle={linkedTextStyle || {}}
        text="Test message:"
        textProps={{ id: 'testText', defaultMessage: 'Test message.' }}
        textStyle={textStyle || {}}
        onClick={() => console.info('Clicked on test link!')}
      />
    </IntlProvider>,
  )

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

const linkedTextStyle = {
  textDecoration: 'underline',
}

const textStyle = { margin: { horizontal: 'medium' } }

test('it renders, has styles and matches snapshot', () => {
  const { container } = renderComponent(linkedTextStyle, textStyle)
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <TextPartialLink
        linkedText={{
          id: `plainButtonLabel`,
          defaultMessage: 'Test Link',
        }}
        text="Test message: "
        textProps={{ id: 'testText', defaultMessage: 'Test message.' }}
        onClick={() => console.info('Clicked on test link!')}
      />
    </IntlProvider>,
  )
  expect(spy).not.toHaveBeenCalled()
})
