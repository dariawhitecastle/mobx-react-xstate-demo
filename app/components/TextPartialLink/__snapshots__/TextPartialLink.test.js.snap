// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <div>
    <div>
      <span
        class="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        Test message.
      </span>
      <button
        class="StyledButton-sc-323bzc-0 kHrfdt"
        type="button"
      >
        Test Link
      </button>
    </div>
  </div>
</div>
`;

exports[`it renders, has styles and matches snapshot 1`] = `
<div>
  <div>
    <div>
      <span
        class="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        Test message.
      </span>
      <button
        class="StyledButton-sc-323bzc-0 kHrfdt"
        style="text-decoration: underline;"
        type="button"
      >
        Test Link
      </button>
    </div>
  </div>
</div>
`;
